frontend/ui/widget/inputtext.lua

```diff
@@ -233,17 +233,18 @@ local function initTouchEvents()
                         },
                         {
                             {
-                                text = _("Cancel"),
+                                text = _("Delete all"),
```

Comment:
I guess I don't mind losing Cancel, as we have a top right `x`.
But I dunno if this is worth it, feels a bit too dedicated: delete all or nothing :). If I have 10 lines among 30 I want to delete, this won't help.
Feels like using `Select`, select start then end, and then being able to Delete what's been selected (not sure how this could behave, as currently it shows only notifications all along) would feel more helpful.

Reply:
> as we have a top right `x`.

And tapping outside, as usual.



> and then being able to Delete what's been selected

The problem is that we do not highlight the selection and must proceed blindly.
Some further improvements of the Clipboard dialog can be considered in the future.

Reply:
> I've often had the need to just wipe the document and start from scratch, but I couldn't find any way of doing this.

https://github.com/koreader/koreader/issues/11161#issuecomment-1827692855

Reply:
FWIW, if the keyboard is open, long press on the Delete key will delete the full *line* (IIRC, it's line by line).

Reply:
I think we call that the backspace key, but yes.

Reply:
The long-press backspace is useful, but would it possible to pop up (and stick before TapClose) for choices between deletion by line, by punctuation, and by word? That would seem to be a nice feature.

Reply:
> long-pressing the `Reset` button

Not available for long notes cleaning.

Reply:
I see. Then I would +1 the above suggestion that long-press on backspace opens a sticky buttondialog with delete buttons for word, punctuation, line, and entire document?

Reply:
>  I would +1 the above suggestion that long-press on backspace opens a sticky buttondialog

-1 to that: I don't think we should mess with keyboard tap & long-press, people may be used to long-press backspace to empty current line, they shouldn't be bothered by a buttondialog popping up and interrupting the keyboard finger dance.
If a popup, it should be a keypopup like with all the other keys (so, like a sticky buttondialog :) but with style !), allowing for swipes as shortcuts to trigger alternate actions. But as the backspace key is stuck to the screen right border, that popup ~~may be shifted left when it appears~~ - looks like no, we can have half popups, allowing for 5 or 7 keys, where some swipes directions could make sense (swipe south to delete to bottom, swipe south west to delete to paragraph end).

Reply:
-1 from me too, for the same reasons, as well as user familiarity with other touch screen keyboards.

Reply:
>> long-pressing the Reset button

>Not available for long notes cleaning.

Couldn't it just delete the file and recreate a blank file with same filename and location?

Reply:
How about calling on the combination of shift + backspace? We can have a rule that a backspace with the shift on won't automatically turn it off (which is currently the case, incidentally).  Then the tap and long-press combinations will offer four free slots for use. It can extend without interfering with user habits.